subxt/src/backend/unstable/mod.rs

```diff
@@ -577,29 +559,27 @@ impl<T: Config + Send + Sync + 'static> Backend<T> for UnstableBackend<T> {
                         };
                         return Poll::Ready(Some(Ok(ev)));
                     } else {
-                        // Keep waiting for more finalized blocks until we find it (get rid of any other block refs
-                        // now, since none of them were what we were looking for anyway).
+                        // Not found it! If follow ev is pending, then return pending here and wait for
+                        // a new one to come in, else loop around and see if we get another one immediately.
```

Comment:
nit: We could probably remove the else blocks to get rid of some extra indentation; up to you 👍

Reply:
I did actually think abotu this, but you'd end up with a bit much for my liking in the new `else` thing eg

```rust
let Some((SeenBlockMarker::Finalized, block_ref)) = seen_blocks.remove(hash) else {
    seen_blocks.clear();
    if follow_ev_is_pending {
        return Poll::Pending;
    } else {
        continue;
    }
};

// ..
```

Unless you had another idea? :)